pkg/component/types.go

```diff
@@ -34,7 +34,19 @@ type ComponentList struct {
 // ComponentStatus is Status of components
 type ComponentStatus struct {
 	Context          string              `json:"context,omitempty"`
-	State            string              `json:"state"`
+	State            State               `json:"state"`
 	LinkedComponents map[string][]string `json:"linkedComponents,omitempty"`
 	LinkedServices   []string            `json:"linkedServices,omitempty"`
 }
+
+// State reperesents component state
+type State string
+
+const (
+	// StateTypePushed means that Storage is present both locally and on cluster
+	StateTypePushed State = "Pushed"
```

Comment:
Shouldn't this be `StateTypePushed = "Pushed"`?

Reply:
`state` here signifies that all constants are of `type state`

Reply:
That's a cool feature. TIL!